src/src/rx_main.cpp

```diff
@@ -650,23 +652,43 @@ void ICACHE_RAM_ATTR ProcessRFPacket()
 
     uint16_t inCRC = ( ( (uint16_t)(Radio.RXdataBuffer[0] & 0b11111100) ) << 6 ) | Radio.RXdataBuffer[7];
 
-    Radio.RXdataBuffer[0] = type;
-    uint16_t calculatedCRC = ota_crc.calc(Radio.RXdataBuffer, 7, CRCInitializer);
-
-    if (inCRC != calculatedCRC)
+    // artificially inject the nonce on data packets for the CRC calculation
+    int foundNonceIndex = -1;
+    for (int index=0 ; index<3 ; index++)
     {
-        #ifndef DEBUG_SUPPRESS
+        uint8_t tryNonce = (NonceRX + nonceOffset[index]) % ExpressLRS_currAirRate_Modparams->FHSShopInterval;
+        Radio.RXdataBuffer[0] = type | (tryNonce << 2);
+        uint16_t crc = ota_crc.calc(Radio.RXdataBuffer, 7, CRCInitializer);
+        if (crc == inCRC)
+        {
+            if (NonceRX / ExpressLRS_currAirRate_Modparams->FHSShopInterval == (NonceRX + nonceOffset[index]) / ExpressLRS_currAirRate_Modparams->FHSShopInterval)
```

Comment:
This conditional confuses me. I'm probably just not awake yet, but assuming the tested offsets are less than fhssinterval it looks like it's just testing for offset 0 and never doing any adjustment. Probably needs a comment to explain what it's actually doing

Reply:
Ah nuts,I read it as modulo not divide. Cool.